samples/jobs/describe_job_execution/main.cpp

```diff
@@ -226,17 +226,20 @@ int main(int argc, char *argv[])
 
         // This isn't absolutely neccessary but since we're doing a publish almost immediately afterwards,
         // to be cautious make sure the subscribe has finished before doing the publish.
+        std::atomic<bool> subAcked(false);
```

Comment:
suggestion: C++ offers more synchronization primitives than aws-c-common offers.
We could use something simpler than this mutex/condition_variable/atomic combo, like:
https://en.cppreference.com/w/cpp/thread/promise
```
std::promise<void> subAcked;

subAcked.set_value(); // to notify as done

subAcked.get_future().wait(); // to wait on it

Reply:
I am stuck by converting stuff like `conditionVariable.wait(uniqueLock, [&]() { return subAcked || jobExecutionAcceptedCompleted; });` to promise... It seems like `when_any(futures...)` is not supported by C++ 11.

Reply:
A possible workaround I can think of is 
`conditionVariable.wait(uniqueLock, [&]() { return connectionSucceeded || connectionClosed; });` -> `if(connectionCompletedPromise.get_future().get() || connectionClosedPromise.get_future().get()){ ... }` ?